torchio/transforms/preprocessing/intensity/projection.py

```diff
@@ -0,0 +1,158 @@
+            If ``False``, some slabs may not be ``slab_thickness`` thick
+            depending on the size of the image, slab thickness, and stride.
+
+    Example:
+        >>> import torchio as tio
+        >>> ct = tio.datasets.Slicer('CTChest').CT_chest
+        >>> axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        >>> ct_t = axial_mips(ct)
+        >>> ct_t.plot()
+
+    .. plot::
+
+        import torchio as tio
+        sub = tio.datasets.Slicer('CTChest')
+        ct = sub.CT_chest
+        axial_mips = tio.Projection("S", slab_thickness=20, stride=20)
+        ct_mips = axial_mips(ct)
+        sub.add_image(ct_mips, 'MIP')
+        sub.plot()
+
+    """
+    def __init__(
+            self,
+            axis: str,
+            slab_thickness: Optional[int] = None,
+            stride: Optional[int] = 1,
+            projection_type: Optional[str] = 'max',
+            q: Optional[float] = None,
+            full_slabs_only: Optional[bool] = True,
+            **kwargs
+            ):
+        super().__init__(**kwargs)
+        self.args_names = (
+            'axis', 'slab_thickness', 'stride',
+            'projection_type', 'q', 'full_slabs_only'
+            )
+        self.axis = axis
+        self.slab_thickness = slab_thickness
+        self.stride = stride
+        self.projection_type = projection_type
+        self.q = q
+        self.full_slabs_only = full_slabs_only
+        self.projection_fun = self.get_projection_function()
+
+    def get_projection_function(self):
+        if self.projection_type == 'max':
+            projection_fun = torch.amax
+        elif self.projection_type == 'min':
+            projection_fun = torch.amin
+        elif self.projection_type == 'mean':
+            projection_fun = torch.mean
+        elif self.projection_type == 'median':
+            projection_fun = torch.median
+        elif self.projection_type == 'quantile':
+            projection_fun = torch.quantile
+            self.validate_quantile()
+        else:
+            message = (
+                '`projection_type` must be one of "max", "min", "mean",'
+                ' "median", or "quantile".'
+                )
+            raise ValueError(message)
+        return projection_fun
+
+    def validate_quantile(self):
+        message = (
+            'For `projection_type="quantile"`, `q` must be a scalar value'
+            f'in the range [0, 1], not {self.q}.'
+            )
+        if self.q is None:
+            raise ValueError(message)
+        elif 0 <= self.q <= 1:
+            pass
+        else:
+            raise ValueError(message)
+
+    def apply_transform(self, subject: Subject) -> Subject:
+        for image in self.get_images(subject):
+            self.apply_projection(image)
+        return subject
+
+    def apply_projection(self, image: ScalarImage) -> None:
+        self.axis_index = image.axis_name_to_index(self.axis)
```

Comment:
*Maybe `axis_name_to_index` could be modified to just return the input if it is 0, 1 or 2.

Reply:
Note that we would need to add 1 to the axis index, because e.g. 0 would be the second dimension (as the channels one is actually the first one).

Reply:
Also, the `spatial_shape` property of the image might be useful.

Reply:
Do you want me to take a look at modifying `axis_name_to_index()`, or just implement a method in `Projection` to validate the chosen axis exists and add 1 to it for indexing purposes?

Reply:
I'll think of the best approach and maybe commit something.